{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "sized-estonia",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: nextstrain [-h]\n",
      "                  {build,view,deploy,remote,shell,update,check-setup,version}\n",
      "                  ...\n",
      "\n",
      "Nextstrain command-line interface (CLI)\n",
      "\n",
      "The `nextstrain` program and its subcommands aim to provide a consistent way to\n",
      "run and visualize pathogen builds and access Nextstrain components like Augur\n",
      "and Auspice across computing environments such as Docker, Conda, and AWS Batch.\n",
      "\n",
      "optional arguments:\n",
      "  -h, --help            show this help message and exit\n",
      "\n",
      "commands:\n",
      "  {build,view,deploy,remote,shell,update,check-setup,version}\n",
      "    build               Run pathogen build\n",
      "    view                View pathogen build\n",
      "    deploy              Deploy pathogen build\n",
      "    remote              Upload, download, and manage Nextstrain files on\n",
      "                        remote sources.\n",
      "    shell               Start a new shell in the build environment\n",
      "    update              Update your local image copy\n",
      "    check-setup         Test your local setup\n",
      "    version             Show version information\n"
     ]
    }
   ],
   "source": [
    "!nextstrain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "specified-kuwait",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: augur [-h]\n",
      "             {parse,filter,mask,align,tree,refine,ancestral,translate,reconstruct-sequences,clades,traits,sequence-traits,lbi,distance,titers,frequencies,export,validate,version,import}\n",
      "             ...\n",
      "\n",
      "Augur: A bioinformatics toolkit for phylogenetic analysis.\n",
      "\n",
      "positional arguments:\n",
      "  {parse,filter,mask,align,tree,refine,ancestral,translate,reconstruct-sequences,clades,traits,sequence-traits,lbi,distance,titers,frequencies,export,validate,version,import}\n",
      "    parse               Parse delimited fields from FASTA sequence names into\n",
      "                        a TSV and FASTA file.\n",
      "    filter              Filter and subsample a sequence set.\n",
      "    mask                Mask specified sites from a VCF or FASTA file.\n",
      "    align               Align multiple sequences from FASTA.\n",
      "    tree                Build a tree using a variety of methods.\n",
      "    refine              Refine an initial tree using sequence metadata.\n",
      "    ancestral           Infer ancestral sequences based on a tree.\n",
      "    translate           Translate gene regions from nucleotides to amino\n",
      "                        acids.\n",
      "    reconstruct-sequences\n",
      "                        Reconstruct alignments from mutations inferred on the\n",
      "                        tree\n",
      "    clades              Assign clades to nodes in a tree based on amino-acid\n",
      "                        or nucleotide signatures.\n",
      "    traits              Infer ancestral traits based on a tree.\n",
      "    sequence-traits     Annotate sequences based on amino-acid or nucleotide\n",
      "                        signatures.\n",
      "    lbi                 Calculate LBI for a given tree and one or more sets of\n",
      "                        parameters.\n",
      "    distance            Calculate the distance between sequences across entire\n",
      "                        genes or at a predefined subset of sites.\n",
      "    titers              Annotate a tree with actual and inferred titer\n",
      "                        measurements.\n",
      "    frequencies         infer frequencies of mutations or clades\n",
      "    export              Export JSON files suitable for visualization with\n",
      "                        auspice.\n",
      "    validate            Validate files related to augur consumption or export.\n",
      "    version             Print the version of augur.\n",
      "    import              Import analyses into augur pipeline from other systems\n",
      "\n",
      "optional arguments:\n",
      "  -h, --help            show this help message and exit\n"
     ]
    }
   ],
   "source": [
    "!augur -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "casual-planet",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# packages in environment at /home/yzu/miniconda3/envs/ns:\r\n",
      "#\r\n",
      "# Name                    Version                   Build  Channel\r\n",
      "_libgcc_mutex             0.1                 conda_forge    conda-forge\r\n",
      "_openmp_mutex             4.5                       1_gnu    conda-forge\r\n",
      "aioeasywebdav             2.4.0           py36h9f0ad1d_1001    conda-forge\r\n",
      "aiohttp                   3.7.3            py36h27cfd23_1  \r\n",
      "appdirs                   1.4.4                      py_0  \r\n",
      "argon2-cffi               20.1.0           py36h7b6447c_1  \r\n",
      "async-timeout             3.0.1            py36h06a4308_0  \r\n",
      "async_generator           1.10             py36h28b3542_0  \r\n",
      "attrs                     20.3.0             pyhd3eb1b0_0  \r\n",
      "augur                     11.0.0                     py_0    bioconda\r\n",
      "bcbio-gff                 0.6.6              pyh864c0ab_1    bioconda\r\n",
      "bcrypt                    3.2.0            py36h7b6447c_0  \r\n",
      "biopython                 1.76                     pypi_0    pypi\r\n",
      "blas                      1.0                         mkl  \r\n",
      "bleach                    1.4.2                    py36_0    bioconda\r\n",
      "boto3                     1.9.66                   py36_0  \r\n",
      "botocore                  1.12.67                  py36_0  \r\n",
      "bx-python                 0.8.9            py36h5e0341f_2    bioconda\r\n",
      "bzip2                     1.0.8                h7b6447c_0  \r\n",
      "ca-certificates           2021.1.19            h06a4308_0  \r\n",
      "cachetools                4.2.1              pyhd3eb1b0_0  \r\n",
      "cairo                     1.14.12              h8948797_3  \r\n",
      "certifi                   2020.12.5        py36h06a4308_0  \r\n",
      "cffi                      1.14.4           py36h261ae71_0  \r\n",
      "chardet                   3.0.4           py36h06a4308_1003  \r\n",
      "conda                     4.9.2            py36h06a4308_0  \r\n",
      "conda-package-handling    1.7.2            py36h03888b9_0  \r\n",
      "configargparse            1.2.3                      py_0  \r\n",
      "cryptography              3.3.1            py36h3c74f83_0  \r\n",
      "cvxopt                    1.2.0            py36hfa32c7d_0  \r\n",
      "cycler                    0.10.0                   py36_0  \r\n",
      "datrie                    0.8.2            py36h27cfd23_0  \r\n",
      "dbus                      1.13.18              hb2f20db_0  \r\n",
      "decorator                 4.4.2                      py_0  \r\n",
      "defusedxml                0.6.0                      py_0  \r\n",
      "docutils                  0.16                     py36_1  \r\n",
      "dropbox                   10.10.0          py36h06a4308_0  \r\n",
      "entrypoints               0.3                      py36_0  \r\n",
      "expat                     2.2.10               he6710b0_2  \r\n",
      "fasttree                  2.1.10               h516909a_4    bioconda\r\n",
      "ffmpeg                    4.0                  hcdf2ecd_0  \r\n",
      "filechunkio               1.6                      py36_0    bioconda\r\n",
      "fontconfig                2.13.0               h9420a91_0  \r\n",
      "freeglut                  3.0.0                hf484d3e_5  \r\n",
      "freetype                  2.10.4               h5ab3b9f_0  \r\n",
      "ftputil                   3.2                      py36_0    bioconda\r\n",
      "gettext                   0.19.8.1             h9b4dc7a_1  \r\n",
      "git                       2.23.0          pl526hacde149_0  \r\n",
      "gitdb                     4.0.5                      py_0  \r\n",
      "gitpython                 3.1.12             pyhd3eb1b0_1  \r\n",
      "glib                      2.66.1               h92f7085_0  \r\n",
      "glpk                      4.65                 h3ceedfd_2  \r\n",
      "gmp                       6.1.2                h6c8ec71_1  \r\n",
      "google-api-core           1.22.2                   py36_0  \r\n",
      "google-auth               1.24.0             pyhd3eb1b0_0  \r\n",
      "google-cloud-core         1.5.0              pyhd3eb1b0_0  \r\n",
      "google-cloud-storage      1.35.0             pyhd3eb1b0_0  \r\n",
      "google-crc32c             1.1.1            py36h27cfd23_0  \r\n",
      "google-resumable-media    1.2.0              pyhd3eb1b0_1  \r\n",
      "googleapis-common-protos  1.52.0                   py36_0  \r\n",
      "graphite2                 1.3.14               h23475e2_0  \r\n",
      "graphviz                  2.38.0            hcf1ce16_1009    conda-forge\r\n",
      "gsl                       2.4                  h14c3975_4  \r\n",
      "gst-plugins-base          1.14.0               h8213a91_2  \r\n",
      "gstreamer                 1.14.0               h28cd5cc_2  \r\n",
      "harfbuzz                  1.8.8                hffaf4a1_0  \r\n",
      "hdf5                      1.10.2               hba1933b_1  \r\n",
      "html5lib                  1.1                        py_0  \r\n",
      "icu                       58.2                 he6710b0_3  \r\n",
      "idna                      2.10               pyhd3eb1b0_0  \r\n",
      "idna_ssl                  1.1.0            py36h06a4308_0  \r\n",
      "importlib-metadata        2.0.0                      py_1  \r\n",
      "importlib_metadata        2.0.0                         1  \r\n",
      "intel-openmp              2020.2                      254  \r\n",
      "ipykernel                 5.3.4            py36h5ca1d4c_0  \r\n",
      "ipython                   5.8.0                    py36_1    conda-forge\r\n",
      "ipython_genutils          0.2.0              pyhd3eb1b0_1  \r\n",
      "iqtree                    2.0.3                h176a8bc_1    bioconda\r\n",
      "jasper                    2.0.14               h07fcdf6_1  \r\n",
      "jinja2                    2.11.2             pyhd3eb1b0_0  \r\n",
      "jmespath                  0.10.0                     py_0  \r\n",
      "jpeg                      9d                   h36c2ea0_0    conda-forge\r\n",
      "json5                     0.9.5                      py_0  \r\n",
      "jsonschema                3.2.0                      py_2  \r\n",
      "jupyter-contrib-core      0.3.3                    pypi_0    pypi\r\n",
      "jupyter-contrib-nbextensions 0.5.1                    pypi_0    pypi\r\n",
      "jupyter-highlight-selected-word 0.2.0                    pypi_0    pypi\r\n",
      "jupyter-latex-envs        1.4.6                    pypi_0    pypi\r\n",
      "jupyter-nbextensions-configurator 0.4.1                    pypi_0    pypi\r\n",
      "jupyter_client            6.1.7                      py_0  \r\n",
      "jupyter_core              4.7.0            py36h06a4308_0  \r\n",
      "jupyterlab                2.2.6                      py_0  \r\n",
      "jupyterlab_pygments       0.1.2                      py_0  \r\n",
      "jupyterlab_server         1.2.0                      py_0  \r\n",
      "kiwisolver                1.3.0            py36h2531618_0  \r\n",
      "krb5                      1.18.2               h173b8e3_0  \r\n",
      "lcms2                     2.11                 h396b838_0  \r\n",
      "ld_impl_linux-64          2.33.1               h53a641e_7  \r\n",
      "libarchive                3.5.1                h899b81a_0    conda-forge\r\n",
      "libcrc32c                 1.1.1                he6710b0_2  \r\n",
      "libcurl                   7.71.1               h20c2e04_1  \r\n",
      "libedit                   3.1.20191231         h14c3975_1  \r\n",
      "libffi                    3.3                  he6710b0_2  \r\n",
      "libgcc-ng                 9.3.0               h2828fa1_18    conda-forge\r\n",
      "libgfortran-ng            7.3.0                hdf63c60_0  \r\n",
      "libglu                    9.0.0                hf484d3e_1  \r\n",
      "libgomp                   9.3.0               h2828fa1_18    conda-forge\r\n",
      "libiconv                  1.16                 h516909a_0    conda-forge\r\n",
      "libopencv                 3.4.2                hb342d67_1  \r\n",
      "libopus                   1.3.1                h7b6447c_0  \r\n",
      "libpng                    1.6.37               hbc83047_0  \r\n",
      "libprotobuf               3.13.0.1             hd408876_0  \r\n",
      "libsodium                 1.0.18               h7b6447c_0  \r\n",
      "libsolv                   0.7.17               h780b84a_0    conda-forge\r\n",
      "libssh2                   1.9.0                h1ba5d50_1  \r\n",
      "libstdcxx-ng              9.3.0               h6de172a_18    conda-forge\r\n",
      "libtiff                   4.1.0                h2733197_1  \r\n",
      "libtool                   2.4.6             h7b6447c_1005  \r\n",
      "libuuid                   1.0.3                h1bed415_2  \r\n",
      "libvpx                    1.7.0                h439df22_0  \r\n",
      "libxcb                    1.14                 h7b6447c_0  \r\n",
      "libxml2                   2.9.10               hb55368b_3  \r\n",
      "lxml                      4.6.2                    pypi_0    pypi\r\n",
      "lz4-c                     1.9.2                heb0550a_3  \r\n",
      "lzo                       2.10                 h7b6447c_2  \r\n",
      "mafft                     7.475                h516909a_0    bioconda\r\n",
      "mamba                     0.7.10           py36h05d92e0_0    conda-forge\r\n",
      "markupsafe                1.1.1            py36h7b6447c_0  \r\n",
      "matplotlib                3.3.2                h06a4308_0  \r\n",
      "matplotlib-base           3.3.2            py36h817c723_0  \r\n",
      "metis                     5.1.0                hf484d3e_4  \r\n",
      "mistune                   0.8.4            py36h7b6447c_0  \r\n",
      "mkl                       2020.2                      256  \r\n",
      "mkl-service               2.3.0            py36he8ac12f_0  \r\n",
      "mkl_fft                   1.2.0            py36h23d657b_0  \r\n",
      "mkl_random                1.1.1            py36h0573a6f_0  \r\n",
      "multidict                 4.7.6            py36h7b6447c_1  \r\n",
      "nbclient                  0.5.1                      py_0  \r\n",
      "nbconvert                 6.0.7                    py36_0  \r\n",
      "nbformat                  5.1.2              pyhd3eb1b0_1  \r\n",
      "ncurses                   6.2                  he6710b0_1  \r\n",
      "nest-asyncio              1.4.3              pyhd3eb1b0_0  \r\n",
      "netifaces                 0.10.9                   pypi_0    pypi\r\n",
      "networkx                  2.5                        py_0  \r\n",
      "nextstrain-cli            2.0.0.post1              pypi_0    pypi\r\n",
      "nodejs                    10.13.0              he6710b0_0  \r\n",
      "notebook                  6.2.0            py36h06a4308_0  \r\n",
      "numpy                     1.19.2           py36h54aff64_0  \r\n",
      "numpy-base                1.19.2           py36hfa32c7d_0  \r\n",
      "olefile                   0.46                     py36_0  \r\n",
      "opencv                    3.4.2            py36h6fd60c2_1  \r\n",
      "openssl                   1.1.1i               h27cfd23_0  \r\n",
      "packaging                 20.8               pyhd3eb1b0_0  \r\n",
      "pandas                    1.1.3            py36he6710b0_0  \r\n",
      "pandoc                    2.11                 hb0f4dca_0  \r\n",
      "pandocfilters             1.4.3            py36h06a4308_1  \r\n",
      "pango                     1.40.14              he752989_2    conda-forge\r\n",
      "paramiko                  2.7.2                      py_0  \r\n",
      "pcre                      8.44                 he6710b0_0  \r\n",
      "perl                      5.26.2               h14c3975_0  \r\n",
      "pexpect                   4.8.0              pyhd3eb1b0_3  \r\n",
      "pickleshare               0.7.5           pyhd3eb1b0_1003  \r\n",
      "pillow                    8.1.0            py36he98fc37_0  \r\n",
      "pip                       20.3.3           py36h06a4308_0  \r\n",
      "pixman                    0.40.0               h7b6447c_0  \r\n",
      "prettytable               2.0.0              pyhd8ed1ab_0    conda-forge\r\n",
      "prometheus_client         0.9.0              pyhd3eb1b0_0  \r\n",
      "prompt_toolkit            1.0.15                     py_1    conda-forge\r\n",
      "protobuf                  3.13.0.1         py36he6710b0_1  \r\n",
      "psutil                    5.7.2            py36h7b6447c_0  \r\n",
      "ptyprocess                0.7.0              pyhd3eb1b0_2  \r\n",
      "py-opencv                 3.4.2            py36hb342d67_1  \r\n",
      "pyasn1                    0.4.8                      py_0  \r\n",
      "pyasn1-modules            0.2.8                      py_0  \r\n",
      "pycosat                   0.6.3            py36h27cfd23_0  \r\n",
      "pycparser                 2.20                       py_2  \r\n",
      "pygments                  2.7.4              pyhd3eb1b0_0  \r\n",
      "pygraphviz                1.3.1                    py36_0    bioconda\r\n",
      "pynacl                    1.4.0            py36h7b6447c_1  \r\n",
      "pyopenssl                 20.0.1             pyhd3eb1b0_1  \r\n",
      "pyparsing                 2.4.7              pyhd3eb1b0_0  \r\n",
      "pyqt                      5.9.2            py36h05f1152_2  \r\n",
      "pyrsistent                0.17.3           py36h7b6447c_0  \r\n",
      "pysftp                    0.2.9                    py36_0    bioconda\r\n",
      "pysocks                   1.7.1            py36h06a4308_0  \r\n",
      "python                    3.6.12               hcff3b4d_2  \r\n",
      "python-dateutil           2.8.1                      py_0  \r\n",
      "python-irodsclient        0.8.2                      py_0    conda-forge\r\n",
      "python-lzo                1.12            py36hbaba66d_1003    conda-forge\r\n",
      "python_abi                3.6                     1_cp36m    conda-forge\r\n",
      "pytz                      2020.5             pyhd3eb1b0_0  \r\n",
      "pyyaml                    5.4.1            py36h27cfd23_1  \r\n",
      "pyzmq                     20.0.0           py36h2531618_1  \r\n",
      "qt                        5.9.7                h5867ecd_1  \r\n",
      "ratelimiter               1.2.0                   py_1002    conda-forge\r\n",
      "raxml                     8.2.12               h516909a_2    bioconda\r\n",
      "readline                  8.0                  h7b6447c_0  \r\n",
      "reproc                    14.2.1               h36c2ea0_0    conda-forge\r\n",
      "reproc-cpp                14.2.1               h58526e2_0    conda-forge\r\n",
      "requests                  2.25.1             pyhd3eb1b0_0  \r\n",
      "rsa                       3.1.4                    py36_0    bioconda\r\n",
      "ruamel_yaml               0.15.87          py36h7b6447c_1  \r\n",
      "s3transfer                0.1.13                   py36_0  \r\n",
      "scipy                     1.5.2            py36h0b6359f_0  \r\n",
      "send2trash                1.5.0              pyhd3eb1b0_1  \r\n",
      "setuptools                52.0.0           py36h06a4308_0  \r\n",
      "simplegeneric             0.8.1                    py36_2  \r\n",
      "sip                       4.19.8           py36hf484d3e_0  \r\n",
      "six                       1.15.0           py36h06a4308_0  \r\n",
      "smmap                     4.0.0              pyhd3eb1b0_0  \r\n",
      "snakemake                 5.5.4                         1    bioconda\r\n",
      "snakemake-minimal         5.5.4                      py_2    bioconda\r\n",
      "sqlite                    3.33.0               h62c20be_0  \r\n",
      "suitesparse               5.2.0                h9e4a6bb_0  \r\n",
      "tbb                       2020.3               hfd86e86_0  \r\n",
      "terminado                 0.9.2            py36h06a4308_0  \r\n",
      "testpath                  0.4.4                      py_0  \r\n",
      "tk                        8.6.10               hbc83047_0  \r\n",
      "tornado                   6.1              py36h27cfd23_0  \r\n",
      "tqdm                      4.7.2                    py36_0    bioconda\r\n",
      "traitlets                 4.3.3                    py36_0  \r\n",
      "treetime                  0.8.1              pyh3252c3a_0    bioconda\r\n",
      "typing-extensions         3.7.4.3              hd3eb1b0_0  \r\n",
      "typing_extensions         3.7.4.3            pyh06a4308_0  \r\n",
      "urllib3                   1.24.3                   py36_0  \r\n",
      "vcftools                  0.1.16               he513fc3_4    bioconda\r\n",
      "wcwidth                   0.2.5                      py_0  \r\n",
      "webencodings              0.5.1                    py36_1  \r\n",
      "wheel                     0.36.2             pyhd3eb1b0_0  \r\n",
      "wrapt                     1.12.1           py36h7b6447c_1  \r\n",
      "xmlrunner                 1.7.7                      py_0    conda-forge\r\n",
      "xorg-kbproto              1.0.7             h7f98852_1002    conda-forge\r\n",
      "xorg-libice               1.0.10               h516909a_0    conda-forge\r\n",
      "xorg-libsm                1.2.2                h470a237_5    conda-forge\r\n",
      "xorg-libx11               1.6.12               h516909a_0    conda-forge\r\n",
      "xorg-libxext              1.3.4                h516909a_0    conda-forge\r\n",
      "xorg-libxpm               3.5.13               h516909a_0    conda-forge\r\n",
      "xorg-libxrender           0.9.10            h516909a_1002    conda-forge\r\n",
      "xorg-libxt                1.1.5             h516909a_1003    conda-forge\r\n",
      "xorg-renderproto          0.11.1            h14c3975_1002    conda-forge\r\n",
      "xorg-xextproto            7.3.0             h7f98852_1002    conda-forge\r\n",
      "xorg-xproto               7.0.31            h27cfd23_1007  \r\n",
      "xz                        5.2.5                h7b6447c_0  \r\n",
      "yaml                      0.2.5                h7b6447c_0  \r\n",
      "yarl                      1.5.1            py36h7b6447c_0  \r\n",
      "zeromq                    4.3.3                he6710b0_3  \r\n",
      "zipp                      3.4.0              pyhd3eb1b0_0  \r\n",
      "zlib                      1.2.11               h7b6447c_3  \r\n",
      "zstd                      1.4.5                h9ceee32_0  \r\n"
     ]
    }
   ],
   "source": [
    "!conda list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "northern-museum",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "using mafft to align via:\n",
      "\tmafft --reorder --anysymbol --nomemsave --adjustdirection --thread 1 Practical1_Data/aligned.fasta.to_align.fasta 1> Practical1_Data/aligned.fasta 2> Practical1_Data/aligned.fasta.log \n",
      "\n",
      "\tKatoh et al, Nucleic Acid Research, vol 30, issue 14\n",
      "\thttps://doi.org/10.1093%2Fnar%2Fgkf436\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# align the sequences\n",
    "!augur align --sequences Practical1_Data/ratites.fasta --output Practical1_Data/aligned.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "derived-radiation",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "aligned.fasta\r\n",
      "aligned.fasta.log\r\n"
     ]
    }
   ],
   "source": [
    "# grep the aligned sequences\n",
    "!ls Practical1_Data/ | grep aligned.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "middle-refrigerator",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Building a tree via:\n",
      "\tiqtree -ninit 2 -n 2 -me 0.05 -nt 1 -s Practical1_Data/aligned-delim.fasta -m GTR  > Practical1_Data/aligned-delim.iqtree.log\n",
      "\tNguyen et al: IQ-TREE: A fast and effective stochastic algorithm for estimating maximum likelihood phylogenies.\n",
      "\tMol. Biol. Evol., 32:268-274. https://doi.org/10.1093/molbev/msu300\n",
      "\n",
      "\n",
      "Building original tree took 0.24700260162353516 seconds\n"
     ]
    }
   ],
   "source": [
    "#Construct the Phylogeny\n",
    "!augur tree --alignment Practical1_Data/aligned.fasta --output Practical1_Data/tree.nwk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "brutal-tunnel",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "*** WARNING: Tree has no internal node names!\n",
      "*** Without internal node names, ancestral sequences can't be linked up to the correct node later.\n",
      "*** If you want to use 'augur export' or `augur translate` later, re-run this command with the output of 'augur refine'.\n",
      "*** If you haven't run 'augur refine', you can add node names to your tree by running:\n",
      "*** augur refine --tree Practical1_Data/tree.nwk --output-tree <filename>.nwk\n",
      "*** And use <filename>.nwk as the tree when running 'ancestral', 'translate', and 'traits'\n",
      "augur ancestral is using TreeTime version 0.8.1\n",
      "\n",
      "Inferred ancestral sequence states using TreeTime:\n",
      "\tSagulenko et al. TreeTime: Maximum-likelihood phylodynamic analysis\n",
      "\tVirus Evolution, vol 4, https://academic.oup.com/ve/article/4/1/vex042/4794731\n",
      "\n",
      "ancestral mutations written to Practical1_Data/nt_muts.json\n"
     ]
    }
   ],
   "source": [
    "#Infer Ancestral Sequences\n",
    "!augur ancestral \\\n",
    "  --tree Practical1_Data/tree.nwk \\\n",
    "  --alignment Practical1_Data/aligned.fasta \\\n",
    "  --output-node-data Practical1_Data/nt_muts.json \\\n",
    "  --inference joint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "nuclear-hollow",
   "metadata": {},
   "outputs": [
    {
     "ename": "IndentationError",
     "evalue": "unexpected indent (<ipython-input-21-bdc02066ebb9>, line 3)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-21-bdc02066ebb9>\"\u001b[0;36m, line \u001b[0;32m3\u001b[0m\n\u001b[0;31m    --output-node-data Practical1_Data/aa_muts.json\u001b[0m\n\u001b[0m    ^\u001b[0m\n\u001b[0;31mIndentationError\u001b[0m\u001b[0;31m:\u001b[0m unexpected indent\n"
     ]
    }
   ],
   "source": [
    "#Identify Amino-Acid Mutations\n",
    "!!augur translate \\\n",
    "  --tree Practical1_Data/tree.nwk \\\n",
    "  --ancestral-sequences Practical1_Data/nt_muts.json \\\n",
    "  --reference-sequence Practical1_Data/emu.fas\n",
    "  --output-node-data Practical1_Data/aa_muts.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "impressed-principal",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
